grakn/service/Session/TransactionService.py

```diff
@@ -36,19 +37,19 @@ def __init__(self, session_id, tx_type, transaction_endpoint):
 
         # open the transaction with an 'open' message
         open_req = RequestBuilder.open_tx(session_id, tx_type)
-        self._communicator.send_receive(open_req)
+        self._communicator.single_request(open_req)
     __init__.__annotations__ = {'tx_type': enums.TxType}
 
     # --- Passthrough targets ---
     # targets of top level Transaction class
 
-    def query(self, query, infer=True):
-        return ResponseReader.ResponseReader.get_query_results(self, Iterator(self._communicator, RequestBuilder.start_iterating_query(query, infer)))
+    def query(self, query, infer=True, batch_size=50):
+        return ResponseReader.ResponseReader.get_query_results(self, Iterator(self._communicator, RequestBuilder.start_iterating_query(query, infer, batch_size)))
```

Comment:
can we split this line out a little? Starting to get hard to read!

Reply:
Sure